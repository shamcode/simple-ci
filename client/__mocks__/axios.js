const METHODS = [
    'get',
    'post',
    'put',
    'delete'
];

const FIXTURES = {};
const MOCKS = {};

function mockMethodFactory( method ) {
    return jest.fn().mockImplementation(
        ( url ) => {
            if ( !FIXTURES[ method ].hasOwnProperty( url ) ) {
                throw new Error( `Missing fixture for ${method.toUpperCase()} url = "${url}"` );
            }
            const { success, data } = FIXTURES[ method ][ url ];
            return Promise[ success ? 'resolve' : 'reject' ]( {
                data
            } );
        }
    );
}

METHODS.forEach(
    method => {
        MOCKS[ method ] = mockMethodFactory( method );
        FIXTURES[ method ] = {};
    }
);

const defaultMocksData = ( () => {
    const data = {};
    data.chain = {
        id: 2,
        name: 'Test chain',
        command: 'echo 1'
    };
    data.project = {
        id: 1,
        name: 'Test',
        cwd: '/tmp/',
        chains: [ data.chain ]
    };
    return data;
} )();

export default {
    defaultMocksData: defaultMocksData,
    mocks: MOCKS,
    create: jest.fn().mockImplementation( () => {
        return {
            ...MOCKS,
            interceptors: {
                request: {
                    use: () => {}
                },
                response: {
                    use: () => {}
                }
            }
        };
    } ),
    use( method, url, data, success = true ) {
        if ( !METHODS.includes( method ) ) {
            throw new Error( `Unknown method: "${method}"`);
        }
        FIXTURES[ method ][ url ] = {
            data,
            success
        };
        return this;
    },
    useDefaultMocks() {
        this
            .use( 'get', '/projects', [ defaultMocksData.project ] )
            .use( 'get', '/projects/1', defaultMocksData.project )
            .use( 'get',  '/chains/2', defaultMocksData.chain );
        return this;
    }
};