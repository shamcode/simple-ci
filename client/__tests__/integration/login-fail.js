import setup, { app } from './helpers';
import axios from 'axios';
jest.mock( 'axios' );

beforeEach( () => {
    jest.resetModules();
    jest.clearAllMocks();
    setup( { auth: false } );
} );

it( 'fail login (invalid header)', async() => {
    expect.assertions( 2 );

    const interceptors = {};
    const getMock = jest.fn( () => {
        return Promise.resolve( { headers: {} } ).then(
            response => interceptors.success( response )
        );
    } );
    axios.create.mockImplementation( () => {
        return {
            defaults: {
                headers: {}
            },
            get: getMock,
            interceptors: {
                request: {
                    use: () => {}
                },
                response: {
                    use: ( success, fail ) => {
                        interceptors.success = success;
                        interceptors.fail = fail;
                    }
                }
            },
            post: jest.fn().mockReturnValue( Promise.resolve() )
        };
    } );

    window.location.href = 'http://simple-ci.example.com/';

    await app.start();
    app.checkBody();
    app.form.fill( 'username', 'admin' );
    app.form.fill( 'password', 'pass' );
    await app.form.submit();
    app.checkBody();
} );
