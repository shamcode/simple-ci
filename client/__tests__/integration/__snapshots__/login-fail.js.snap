// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`fail login (invalid header) 1`] = `
"<header>
  <a class=\\"logo\\" href=\\"\\"> Simple CI </a>
</header>
<!--0-->
<main>
  <!--0-->
  <!--0-->
  <div class=\\"login-form\\">
    <form>
      <div class=\\"group\\">
        <input name=\\"username\\" type=\\"text\\" required=\\"\\"><span class=\\"highlight\\"></span><span class=\\"bar\\"></span>
        <label>Username</label>
      </div>
      <div class=\\"group\\">
        <input name=\\"password\\" type=\\"password\\" required=\\"\\"><span class=\\"highlight\\"></span><span class=\\"bar\\"></span>
        <label>Password</label>
      </div>
      <div class=\\"group\\">
        <button type=\\"submit\\"><span>Login</span></button>
      </div>
    </form>
  </div>
</main>
<footer> Sham 2018
  (v 0.0.8) </footer>
<!--1-->"
`;

exports[`fail login (invalid header) 2`] = `
"<header>
  <a class=\\"logo\\" href=\\"\\"> Simple CI </a>
</header>
<!--0-->
<main>
  <ul class=\\"errors-panel\\">
    <li>Login fail</li>
  </ul>
  <!--0-->
  <!--0-->
  <div class=\\"login-form\\">
    <form>
      <div class=\\"group\\">
        <input name=\\"username\\" type=\\"text\\" required=\\"\\"><span class=\\"highlight\\"></span><span class=\\"bar\\"></span>
        <label>Username</label>
      </div>
      <div class=\\"group\\">
        <input name=\\"password\\" type=\\"password\\" required=\\"\\"><span class=\\"highlight\\"></span><span class=\\"bar\\"></span>
        <label>Password</label>
      </div>
      <div class=\\"group\\">
        <button type=\\"submit\\"><span>Login</span></button>
      </div>
    </form>
  </div>
</main>
<footer> Sham 2018
  (v 0.0.8) </footer>
<!--1-->"
`;
